weather.py

```diff
@@ -0,0 +1,414 @@
+        humid_date = humid_date.split("-")
+        highest_month = Weatherman.months[int(highest_date[1]) - 1]
+        highest_year = highest_date[2]
+        lowest_month = Weatherman.months[int(lowest_date[1]) - 1]
+        lowest_year = lowest_date[2]
+        humid_month = Weatherman.months[int(humid_date[1]) - 1]
+        humid_year = humid_date[2]
+
+        return highest_temp, lowest_temp, highest_month, lowest_month, highest_year, lowest_year, humid_percent, humid_month, humid_year
+
+    def average_argument_handler(self, current_argument):
+        date = current_argument.split("/")
+        is_valid = self.validate_month_and_year(date)
+
+        file = self.get_file_path(date[0], date[1])
+
+        if is_valid:
+            is_file = self.file_exists(file)
+
+            if is_file:
+                result = self.calculate_average(file)
+
+                # Function that prints report according to the mode provided
+                self.generate_report_average(result)
+            else:
+                print(f"Files for the date {current_argument} are not in the system.")
+
+    def validate_month_and_year(self, date):
+        if len(date) < 2:
+            print("Please Enter the correct year and month for -c")
+            return False
+        if len(date[0]) != 4:  # To check if year is entered correctly
+            print("Wrong year Format entered")
+            return False
+        if 0 > int(date[1]) < 13:  # Validate month is in between 1-12
+            print("Wrong month Format entered")
+            return False
+        return True
+
+    def file_exists(self, file):
+        if path.exists(file):
+            return True
+        else:
+            return False
+
+    def calculate_average(self, file):
+        avg_highest_temperature = 0
+        avg_lowest_temperature = 0
+        avg_humidity = 0
+
+        count_highest_temperature = 0  # variables to count number of occurrences to calculate avg
+        count_lowest_temperature = 0
+        count_humidity = 0
+
+        if path.exists(file):
+            reader = CSVReader(file)
+
+            record = reader.get_record()
+            while record:
+                avg_highest_temperature, count_highest_temperature, avg_lowest_temperature, \
+                 count_lowest_temperature, avg_humidity, count_humidity \
+                 = self.calculate_average_helper(record, avg_highest_temperature, count_highest_temperature,
+                                                 avg_lowest_temperature, count_lowest_temperature, avg_humidity, count_humidity)
+                record = reader.get_record()
+
+        results = dict()
+
+        # Storing average of results obtained in class level dictionaries
+        results["avghighest_temperature"] = avg_highest_temperature / count_highest_temperature
+        results["avglowest_temperature"] = avg_lowest_temperature / count_lowest_temperature
+        results["avghumid"] = avg_humidity / count_humidity
```

Comment:
Here you also can use curly braces dictionary format to save these values into the dictionary.